TLSF: unused parameter warnings

During compilation unused parameter warnings are emitted for TLSF:
```
Compiling C: /vagrant/build/stm32f4_discovery/protothreads/libxpcc/ext/tlsf/tlsf.o
/vagrant/ext/tlsf/tlsf.c: In function 'default_walker':
/vagrant/ext/tlsf/tlsf.c:907:34: warning: unused parameter 'ptr' [-Wunused-parameter]
 static void default_walker(void* ptr, size_t size, int used, void* user)
                                  ^
/vagrant/ext/tlsf/tlsf.c:907:46: warning: unused parameter 'size' [-Wunused-parameter]
 static void default_walker(void* ptr, size_t size, int used, void* user)
                                              ^
/vagrant/ext/tlsf/tlsf.c:907:56: warning: unused parameter 'used' [-Wunused-parameter]
 static void default_walker(void* ptr, size_t size, int used, void* user)
                                                        ^
```

--- a/ext/tlsf/tlsf.c
+++ b/ext/tlsf/tlsf.c
@@ -28,7 +28,8 @@
 ** since xpcc does not use newlib's printf.
 */
 #define tlsf_assert (void)
-#define printf(...) {}
+static void silent_printf(char* fmt, ...) { (void)fmt; }
+#define printf silent_printf
 
 
 #if defined(__cplusplus)
@@ -1271,3 +1272,6 @@ void* tlsf_realloc(tlsf_t tlsf, void* ptr, size_t size)
 
 	return p;
 }
+
+#undef tlsf_assert
+#undef printf

